File: tests/golden/dart/full_module.dart
import 'dart:async';
import 'dart:convert';

// Future

abstract class UserRepository {
  User? findById(String id);

  List<User> findAll();
}

/// In-memory implementation of UserRepository.
class InMemoryUserRepository implements UserRepository {
  final List<User> _users = [];

  @override
  User? findById(String id) {
    return _users.cast<User?>().firstWhere(
      (u) => u?.id == id,
      orElse: () => null,
    );
  }

  @override
  List<User> findAll() {
    return List.unmodifiable(_users);
  }
}

User parseUser(String json) {
  final data = jsonDecode(json);
  return User.fromMap(data);
}

File: tests/golden/kotlin/full_module.kt
import kotlin.collections.ArrayList
import kotlin.collections.List
import kotlin.collections.MutableList

internal interface UserRepository {
    internal fun findById(id: String): User?

    internal fun findAll(): List<User>
}

/**
 * In-memory implementation of UserRepository.
 */
internal class InMemoryUserRepository : UserRepository {
    private val users: MutableList<User>

    internal override fun findById(id: String): User? {
        return users.firstOrNull { it.id == id }
    }

    internal override fun findAll(): List<User> {
        return ArrayList<User>(users)
    }
}
